test/test-vfscanf-percent-a.c

```diff
@@ -0,0 +1,104 @@
+/*
+* SPDX-License-Identifier: BSD-3-Clause
+* 
+* Copyright © 2024, Synopsys Inc.
+* Copyright © 2024, Solid Sands B.V.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions
+* are met:
+*
+* 1. Redistributions of source code must retain the above copyright
+*    notice, this list of conditions and the following disclaimer.
+*
+* 2. Redistributions in binary form must reproduce the above
+*    copyright notice, this list of conditions and the following
+*    disclaimer in the documentation and/or other materials provided
+*    with the distribution.
+*
+* 3. Neither the name of the copyright holder nor the names of its
+*    contributors may be used to endorse or promote products derived
+*    from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+* INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+* STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+* OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include <stdio.h>
+#include <math.h>
+
+#ifndef TEST_FILE_NAME
+#define TEST_FILE_NAME "VFSCANFPERCENTaA.TXT"
+#endif
+
+const char* input_string = "0x1.1p+3 0x1.6p+2 ";
+
+int compare_floats(float a, float b, float tolerance) {
+    return fabs(a - b) < tolerance;
+}
+
+int main() {
+    int res = -1;
+    int ret = 0;
+    float var = 0.0F;
+    float tolerance = 1e-6;
+    const float ref_out = 5.5F;
+
+    FILE *file = NULL;
+
+    file = fopen(TEST_FILE_NAME, "w");
+    if( file == NULL )
+        return 0;
+
+    res = fprintf(file, "%s", input_string);
+
+    fclose(file);
+
+    {
+        file = fopen(TEST_FILE_NAME, "r");
+        if( file == NULL )
+            return 0;
+
+        res = fscanf(file, "%a %a ", &var, &var);
```

Comment:
Re-using the same variable for both results is definitely legal, but I'm not sure we want to be testing that along with the %a/%A stuff? Why not use two separate variables and then validating the result of both conversions?